engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceModificationProcessor.java

```diff
@@ -83,15 +87,45 @@ private void activateElement(
       final AbstractFlowElement elementToActivate) {
     final var elementInstanceRecord = new ProcessInstanceRecord();
     elementInstanceRecord.wrap(processInstance);
+    // todo: deal with non-existing flow scope (#9643)
+    final Optional<Long> flowScopeKey = getFlowScopeKey(processInstance, elementToActivate);
     commandWriter.appendFollowUpCommand(
         keyGenerator.nextKey(),
         ProcessInstanceIntent.ACTIVATE_ELEMENT,
         elementInstanceRecord
-            // todo: allow non-process instance element's as flowscope
-            .setFlowScopeKey(processInstance.getProcessInstanceKey())
+            .setFlowScopeKey(flowScopeKey.get())
             .setBpmnElementType(elementToActivate.getElementType())
             .setElementId(instruction.getElementId())
             .setParentProcessInstanceKey(-1)
             .setParentElementInstanceKey(-1));
   }
+
+  private Optional<Long> getFlowScopeKey(
```

Comment:
🔧 Let's rename this (and the overloaded method) to `findFlowScopeKey`, to highlight that its uncertain whether it can be found